archiver.go

```diff
@@ -113,6 +113,19 @@ func (a *Archiver) do(log log15.Logger, j *Job) (err error) {
 		"last-fetch", r.FetchedAt,
 		"references", len(r.References))
 
+	defer func() {
+		if rcv := recover(); rcv != nil {
+			log.Error("panic while processing repository", "error", rcv)
```

Comment:
It would be nice to also have the stacktrace.

Reply:
I don't know if it's possible, there are the functions `debug.PrintStack()` and `runtime.Stack()`, but they give you the stacktrace from they are called, so we woudn't have the stacktrace from the instruction which raised the problem.

@src-d/data-retrieval does some of you know how get the stacktrace we want in this case?

Reply:
https://play.golang.org/p/wVj32VTnYpH `debug.Stack`